{
  try {
    waiter.get();
  }
 catch (  InterruptedException e) {
    throw new InterruptedIOException();
  }
catch (  ExecutionException e) {
    FutureCallback.rethrow(e);
  }
}
